### PR TITLE
feature/request-factory

### PR DESCRIPTION
**Pull request by pdiazvargas**
Closes
- Create a request factory https://github.com/eabglobal/minik/issues/28
- Add debug flag https://github.com/eabglobal/minik/issues/24
- Support other content types https://github.com/eabglobal/minik/issues/10

### CODE REVIEW

File: minik/core.py [hunk 1]
```diff
@@ -86,35 +104,35 @@ def __call__(self, event, context):
...
 
-        return response.to_dict()
+        return self.response.to_dict()
 
-    def _execute_view(self, view, request):
+    def _execute_view(self, view):
         """
         Given a view function, execute the view with the given uri_parameters as
-        argumetns and return a JsonResponse.
+        argumetns and return a.
```
Comment (bwarren2) [inline minik/core.py]: Nitpick: typo.

Comment (pdiazvargas) [reply minik/core.py]: thank you!

File: tests/test_middleware.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    mock_app = MagicMock(response=Response())
+
+    ServerErrorMiddleware()(mock_app, error)
+
+    assert mock_app.response.status_code == error.status_code
+    assert mock_app.response.body == {'error_message': str(error)}
+
+
+class CustomMiddleware:
+    expected_header_value = 'JuniperRidge'
```
Comment (bwarren2) [inline tests/test_middleware.py]: Cute.

**Comment by bwarren2** [comment]
LGTM.  One architecture question: it looks like all the middleware is response-middleware.  Is that true?  Is that what you want?

**Comment by pdiazvargas** [comment]
@bwarren2 for the time being yes. In the future we can expand this concept to be more comprehensive. For the time being this was a good step in the right direction. Let me know if you have any suggestions or opinions in the matter.

Future enhancements can actually include a router middleware and a middleware that would actually execute the view. As in:

```
    def __call__(self, event, context):
        self.request = self._request_builder.build(event, context)
        self.response = Response(
            status_code=codes.ok,
            headers={'Content-Type': 'application/json'}
        )
        self.route = self._router.find(self.request)

        with error_handling(self):
            for middleware in self._middleware:
                middleware(self)

        return self.response.to_dict()
```

**Comment by bwarren2** [comment]
I think Django's approach is very elegant and more easily extensible to request + response middleware: https://docs.djangoproject.com/en/2.2/topics/http/middleware/ and https://github.com/django/django/blob/bff5ccff7501d7bafb53d6110e311882e716bee5/django/core/handlers/base.py .  But it's also pretty heavy, so we can push it off.

**Comment by pdiazvargas** [comment]
@bwarren2 definitely worth considering a lightweight version of it.

**Pull request merged**